Homework 16b

TestProGio opened the pull request:
1. Create a new branch named homework-16.
 
2. Create a new file, Homework16.java in IntelliJ IDEA.

3. Create a test case, registrationNavigation() using @Test annotation.

4. Using Selenium, navigate to "https://qa.koel.app/".

5. Click the Registration link.

6. Verify that you are redirected to Registration page using Assert method.

7. Commit your changes to the new branch homework-16.

8. Push your code to the remote repository.

TestProGio committed:
First Commit on my Fork

Forked a cloned repo from A59 testPro

TestProGio committed:
homework-16b

Pardon the 'b' but I was starting to think something was wrong with the version I fetched.

Create a new file, Homework16.java in IntelliJ IDEA.

Create a test case, registrationNavigation() using @Test annotation.

Using Selenium, navigate to "https://qa.koel.app/".

Click the Registration link.

Verify that you are redirected to Registration page using Assert method.

Commit your changes to the new branch homework-16.

Push your code to the remote repository.

Berichicko commented:
You got this problem, please correct it.

![image](https://github.com/Test-Pro-Bootcamp/A59-QA-Automation-Framework/assets/104761482/41b1735c-7fb9-43a2-89e2-822035cffba6)

**HINT**

1) How to analyze the results of tests:

Test status: 
![Test status_en](https://github.com/Test-Pro-Bootcamp/A53-QA-Automation-Framework/assets/104761482/bfa74efc-b45b-4ca0-8d89-f5224a51bc66)

	
1) Test error (red). This status is assigned to tests that caused an exception from the tested source code.

The tester needs to identify an issue in their code: Test error. This status is assigned to tests that trigger an exception in the tested source code.
In software testing, the "Test error" status is assigned to tests that trigger an exception in the tested source code. When a test case fails due to an unexpected exception being thrown during its execution, it is marked as a test error. This indicates that the code being tested has not handled the exception properly, and further investigation and debugging are required to address the issue.

This status indicates an error in the code and the tester must fix this error:

![Test error_en](https://github.com/Test-Pro-Bootcamp/A53-QA-Automation-Framework/assets/104761482/701a79a3-7c65-4a84-9180-e221c559ea02)

2) Test failed(yellow). If at least one test receives this status, then all its parents are marked as failed. (BUGS)
![Test failed_en](https://github.com/Test-Pro-Bootcamp/A53-QA-Automation-Framework/assets/104761482/0171eacb-5e3f-4e2a-9a35-e0e626260d1d)

**For example**

**NoSuchElementException**  

Verify the locator: As a first step, you ought to find out what kind of issue you are dealing with. Is the element still there and can't be found, or is it no longer present on the page? To do this, you can setup Selenium to not to close the browser after the test failure, and run the test. Once you get the exception, you can inspect the page that is open in the browser. Double-check that the locator being used is correct and up-to-date. Inspect the HTML source code to confirm that the locator uniquely identifies the target element on the page.


![image](https://github.com/Test-Pro-Bootcamp/A59-QA-Automation-Framework/assets/104761482/d38f1bb9-d0d0-458e-ad34-fabd2decab99)


![image](https://github.com/Test-Pro-Bootcamp/A59-QA-Automation-Framework/assets/104761482/611287a3-9dea-4f26-a96b-17062aa97118)

TestProGio commented:
@Berichicko: I corrected my homework but don't see the input box to paste the new pull request for your viewing. Please confirm I corrected the test successfully. Thank you!

Link to Corrected Test:
https://github.com/Test-Pro-Bootcamp/A59-QA-Automation-Framework/pull/24
